sys/suit/coap.c

```diff
@@ -0,0 +1,508 @@
+        }
+        else {
+            DEBUG("error fetching block\n");
+            res = -1;
+            goto out;
+        }
+
+        num += 1;
+    }
+
+out:
+    sock_udp_close(&sock);
+    return res;
+}
+
+int suit_coap_get_blockwise_url(const char *url,
+                               coap_blksize_t blksize,
+                               coap_blockwise_cb_t callback, void *arg)
+{
+    char hostport[SOCK_HOSTPORT_MAXLEN];
+    char urlpath[SOCK_URLPATH_MAXLEN];
+    sock_udp_ep_t remote;
+
+    if (strncmp(url, "coap://", 7)) {
+        puts("no coap");
+        return -EINVAL;
+    }
+
+    if (sock_urlsplit(url, hostport, urlpath) < 0) {
+        puts("urlsplit");
+        return -EINVAL;
+    }
+
+    if (sock_udp_str2ep(&remote, hostport) < 0) {
+        puts("str2ep");
+        return -EINVAL;
+    }
+
+    if (!remote.port) {
+        remote.port = COAP_PORT;
+    }
+
+    return suit_coap_get_blockwise(&remote, urlpath, blksize, callback, arg);
+}
+
+typedef struct {
+    size_t offset;
+    uint8_t *ptr;
+    size_t len;
+} _buf_t;
+
+static int _2buf(void *arg, size_t offset, uint8_t *buf, size_t len, int more)
+{
+    (void)more;
+
+    _buf_t *_buf = arg;
+    if (_buf->offset != offset) {
+        return 0;
+    }
+    if (len > _buf->len) {
+        return -1;
+    }
+    else {
+        memcpy(_buf->ptr, buf, len);
+        _buf->offset += len;
+        _buf->ptr += len;
+        _buf->len -= len;
+        return 0;
+    }
+}
+
+ssize_t suit_coap_get_blockwise_url_buf(const char *url,
+                               coap_blksize_t blksize,
+                               uint8_t *buf, size_t len)
+{
+    _buf_t _buf = { .ptr=buf, .len=len };
+    int res = suit_coap_get_blockwise_url(url, blksize, _2buf, &_buf);
+    return (res < 0) ? (ssize_t)res : (ssize_t)_buf.offset;
+}
+
+static void _suit_handle_url(const char *url)
+{
+    LOG_INFO("suit_coap: downloading \"%s\"\n", url);
+    ssize_t size = suit_coap_get_blockwise_url_buf(url, COAP_BLOCKSIZE_64, _manifest_buf,
+                                              SUIT_MANIFEST_BUFSIZE);
+    if (size >= 0) {
+        LOG_INFO("suit_coap: got manifest with size %u\n", (unsigned)size);
+
+        riotboot_flashwrite_t writer;
+#ifdef MODULE_SUIT_V4
+        suit_v4_manifest_t manifest;
+        memset(&writer, 0, sizeof(manifest));
+
+        manifest.writer = &writer;
+        manifest.urlbuf = _url;
+        manifest.urlbuf_len = SUIT_URL_MAX;
+
+        int res;
+        if ((res = suit_v4_parse(&manifest, _manifest_buf, size)) != SUIT_OK) {
+            printf("suit_v4_parse() failed. res=%i\n", res);
+            return;
+        }
+
+        printf("suit_v4_parse() success\n");
+        if (!(manifest.state & SUIT_MANIFEST_HAVE_IMAGE)) {
+            puts("manifest parsed, but no image fetched");
+            return;
+        }
+
+        res = suit_v4_policy_check(&manifest);
+        if (res) {
+            return;
+        }
+
+#endif
+        if (res == 0) {
```

Comment:
While reviewing #11805 I realized `res` is uninitialized if `MODULE_SUIT_V4` is not used.